### DISCUSSION TITLE
How to rename a column which is used as part of a foreign key?

### DISCUSSION THREAD

**Discussion by cimp8504** [groue/GRDB.swift#1181] [Q&A] [answered]
Hello, 

I'm a big fan of GRDB and we use it in our app.  I have a question involving how to rename a column that is currently used as part of a foreign key.  

In v1 of our migrations, we have a table and it has a foreign key declaration that looks like this:
````
            table.foreignKey(["accountId", "opco"],
                             references: "account",
                             columns: ["id", "opco"],
                             onDelete: .cascade)
````

In a later migration, I want to rename the "accountId" column.  Renaming a column is easy, but now any references using the FK will break.  Is there a simple way to do this that I am missing?  Or will this require creating a temp table, copying the data, dropping the old table and then renaming the temp table?

Thanks in advance for any help!

**Comment by groue** [comment, marked as answer]
Hello @cimp8504,

You're about to recreate the table indeed. Follow closely the technique described in the [Migrations](https://github.com/groue/GRDB.swift/blob/master/Documentation/Migrations.md) guide, "Advanced Database Schema Changes" chapter, in order to talk to SQLite in a proper way.
